src/Widgets/ArticleViewHeader.vala

```diff
@@ -30,6 +30,15 @@ public class FeedReader.ArticleViewHeader : Gtk.HeaderBar {
 	public signal void popClosed();
 	public signal void popOpened();
 
+	private void onMarkButton()
```

Comment:
I wrapped the signal emition to be able to connect / disconnect the wrapped function. If I try to connect to toggledMarked directly, I get this when compiling:
```
/run/build/feedreader/src/Widgets/ArticleViewHeader.vala:51.33-51.45: error: Argument 1: Cannot convert from `FeedReader.ArticleViewHeader.toggledMarked' to `Gtk.ToggleButton.toggled'
		m_mark_button.toggled.connect(toggledMarked);
		                              ^^^^^^^^^^^^^
```
The connect/disconnect has to happen when the widget is updated from the outside, like from the sidebar for example.